### ISSUE TITLE
Combobox: Binding `value` undesirely expands combobox

### ISSUE THREAD

**Issue by bentolor** [reach/reach-ui#755]
## 🐛 Bug report

Thanks @chaance for this beautifuly library. We are trying to use it in a Form editor page and have an issue with the combobox

### Current Behavior

When we set the current value, the combobox displays the expand state including popup. This even occurs if it is not visible (i.e. being in a collapsed section). Regardless of that, it makes it unusable, as it overlays/hides many other fields in the form.

This is how we use it in React:

```
import { faChevronDown } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  Combobox,
  ComboboxInput,
  ComboboxList,
  ComboboxOption,
  ComboboxPopover,
} from '@reach/combobox'
import '@reach/combobox/styles.css'
import React from 'react'
import Attribute from './Attribute'
import Delete from './shared/Delete'

/**
 * @param {{
 *  label: string
 *  description: string
 *  options: string[]
 *  defaultValue?(): string
 *  required?: boolean
 *  deletable?: boolean
 *  validationErrors: import('../../../../shared/validationTypes').ValidationError[]
 *  dataPath: string
 *  value: unknown
 *  onUpdate({}): void
 * }} props
 */
export default function EnumAttribute({
  options,
  required = true,
  deletable,
  ...props
}) {
  return (
    <Attribute {...props}>
      {({ onChange, onDelete }) => (
        <div className="max-w-md flex">
          <div className="w-full">
            <Combobox
              className="w-full"
              openOnFocus
              onSelect={(item) => {
                onChange(item)
              }}
            >
              <label className="block w-full flex">
                <ComboboxInput
                  value={/** @type {string} */ (props.value)}
                  className="border border-gray-400 py-1 px-2 w-full shadow-inner rounded-l"
                  selectOnClick
                  required={required}
                  onChange={(e) => {
                    onChange(e.target.value)
                  }}
                />
                <div className="flex items-center justify-center w-8 text-xs border border-gray-400 rounded-r bg-white hover:bg-gray-200 cursor-pointer">
                  <FontAwesomeIcon icon={faChevronDown} />
                </div>
              </label>
              <ComboboxPopover>
                <ComboboxList persistSelection>
                  {options.map((option, index) => (
                    <ComboboxOption key={index} value={option} />
                  ))}
                </ComboboxList>
              </ComboboxPopover>
            </Combobox>
          </div>
          {deletable ? (
            <Delete
              doDelete={() => {
                onDelete()
              }}
            />
          ) : null}
        </div>
      )}
    </Attribute>
  )
}

```

### Expected behavior

The combobox should not expand itself unless by user interaction

### Reproducible example

Sorry – I have no minimal reproducible at hand. I hope the issue might be obvious/simple enought by the upper code example.

### Suggested solution(s)

I don't know. Any clue what might cause this behaviour? 
Is it by design? 
Is it a bug?
Is there an easy workaround?

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Reach Package    |  @reach/combobox       |     0.13.0    |
| React            |        react  |  17.0.1       |
| Browser          |    Firefox     |     86.0    |
| Assistive tech   |       ?  |  ?       |
| Node             |    node     |  v14.16        |
| npm/yarn         |  npm       | 6.14         |
| Operating System |      Ubuntu   |  20.04        |

**Comment by bentolor** [comment]
@chaance I updated the description with a specific example. Does anything from the HTML code look phishy?

**Comment by indiesquidge** [comment]
@bentolor it's difficult to debug with static example code, especially when there is more code than Reach playing a part and you're saying its context-specific.

Would you mind including creating a minimal reusable example with the [CodeSandbox Template](https://codesandbox.io/s/reach-ui-template-xonfs) so that we can isolate the issue to Reach to better solve your problem?

**Comment by geekus** [comment]
Don't mean to hijack your thread @bentolor but what you describes sound familiar and is an issue I'm struggling with myself and considered opening an issue for. So please confirm if this replicates your problem or not: https://codesandbox.io/s/reach-combobox-forked-lxxuj?file=/src/App.js

The issue is that changing the value in the input field programmatically by changing the value to the input field opens the popover with suggestions. Preferrably the field would get the new value and the popover did not open. The popover should only open when the field is activated by the user focusing on it, or typing in the field if `openOnFocus` is `false`.

**Comment by bentolor** [comment]
Thanks @indiesquidge and @geekus for your kind feedback!

Yes, @geekus – glad that you jumped in! That looks exactly like our issue! I only wonder, because the issue is not appearing in all of our comboboxes. Maybe due to different points in time, when the value actually is set?

We are already in the bugfixing cycle and the deadline is pressing. Therefore I'll fail to deliver a reproducer. The final (simple) editor will be open-source, so I'll be able to provide the showcase, soon. But I'm pretty confident that @geekus already pinpointed the issue.

Any hint for a quick'n'dirty workaround? Otherwise we'll probably have to go to replace it.  :-(

**Comment by geekus** [comment]
Yes, from my experience it seems like if the value is set before the combobox or input is rendered, and then not changed, it will not open. But if changed after first render it opens the popover.

**Comment by karlwills** [comment]
@bentolor I was just wondering if you managed a workaround for this at all? I'm having a similar issue and was wondering if there was a solution you ran with?

**Comment by bentolor** [comment]
@karlwills Due to time constraints and the fact, the the handling inside the Reach Combobox is managed by a state machine and according to our quick research the place where a fix would be needed no longer can distinguish between "user triggered change" and "programatically triggered change" we reckoned that there won't be a easy 3-liner fix.

Therefore we dropped reach-combobox in favour of `material-ui` project.

**Comment by Rada8732** [comment]
Edit it and retry

**Comment by dlacaille** [comment]
@Rada8732 my PR was not merged yet. It seems like the bug is still occuring.

**Comment by chaance** [comment]
This was addressed in #783, but I actually need to revert that as it introduced some other regressions that I unfortunately didn't discover until after I merged it. I will work on a new fix for this next week.

**Comment by tdurand** [comment]
Sorry to be that person, but any progress on this ?

**Comment by tdurand** [comment]
Any workaround to this, I tried to trigger manually `.blur()` on the input to close the popover then selecting a value, but it does not work ?
